Objects/unicodeobject.c

```diff
@@ -4238,6 +4242,19 @@ unicode_decode_call_errorhandler_writer(
     replen = PyUnicode_GET_LENGTH(repunicode);
     if (replen > 1) {
         writer->min_length += replen - 1;
+        need_to_grow = 1;
+    }
+    new_inptr = *input + newpos;
+    if (*inend - new_inptr > remain) {
+        /* We don't know the decoding algorithm here so we make the worst
```

Comment:
The error handler registered with `codecs.register_error` can produce _any_ string as an output. This output will be inserted in the buffer you allocate. So I propose to expect that decoded output could be bigger and either check that and reallocate the buffer or raise a MemoryError exception with proper description.

Reply:
Thanks for your review @sibiryakov . But the situation you mention has already been handled. See the previous condition `repel - 1`, if the handler returns any string longer than one, `need_to_grow` will be set and we will allocate more memory.

Reply:
Great @zhangyangyu! Thank you!